### PR TITLE
node 20.7.0

### PR DESCRIPTION
**Pull request by lpinca**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/n/node.rb [hunk 1]
```diff
@@ -49,8 +49,8 @@ class Node < Formula
   # We track major/minor from upstream Node releases.
   # We will accept *important* npm patch releases when necessary.
   resource "npm" do
-    url "https://registry.npmjs.org/npm/-/npm-9.8.1.tgz"
-    sha256 "0f3e2ae9569291274114b765d8d79b644f63fa162497daee54446d9a600e0f79"
+    url "https://registry.npmjs.org/npm/-/npm-10.1.0.tgz"
+    sha256 "7cb31c0a881964a22577fd84e5a9a5b11e6f49ef8aa0893036b0b68015056252"
```
Comment (chenrui333) [inline Formula/n/node.rb:53]: sha got changed

Comment (chenrui333) [reply Formula/n/node.rb:53]:
```
Expected: ef4bd7e040bb30e9e03249538ba973e2108e11ec2f1b98a886ff42f67249cdb5
  Actual: 3fcfdcd05c461517480596596674df85b35cfce597dd0ae33f5416fc4df12bea
```

File: Formula/n/node.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class Node < Formula
   desc "Platform built on V8 to build network applications"
   homepage "https://nodejs.org/"
-  url "https://nodejs.org/dist/v20.6.1/node-v20.6.1.tar.xz"
-  sha256 "3aec5e728daa38800c343b129221d3488064a2529a39bb5467bc55be226c6a2b"
+  url "https://nodejs.org/dist/v20.7.0/node-v20.7.0.tar.xz"
+  sha256 "ef4bd7e040bb30e9e03249538ba973e2108e11ec2f1b98a886ff42f67249cdb5"
```
Comment (chenrui333) [inline Formula/n/node.rb:5]:
```
Expected: ef4bd7e040bb30e9e03249538ba973e2108e11ec2f1b98a886ff42f67249cdb5
  Actual: 3fcfdcd05c461517480596596674df85b35cfce597dd0ae33f5416fc4df12bea
```

**Review by chenrui333 — APPROVED** [review]

**Comment by karlhorky** [comment]
Node.js 18.18.0 PR over here (although marked as "formula deprecated"):

- #143899

**Review by branchv — APPROVED** [review]

**Comment by karlhorky** [comment]
Thanks for the PR, review and merge @lpinca @branchvincent @chenrui333 🙌

**Pull request merged**